### ISSUE TITLE
API to hide and show app

### ISSUE THREAD

**Issue by sindresorhus** [atom/electron#3694]
I need ability to programmatically hide the app as if the user did <kbd>Cmd</kbd> <kbd>H</kbd>.

I'm thinking: `app.show()` and `app.hide()`.

**Comment by KevinMartin** [comment]
Couldn't you do something like:

On Darwin:
  Hide: `Menu.sendActionToFirstResponder('hide:');` (works)
  Show: `Menu.sendActionToFirstResponder('show:');` (haven't tested)
On other platforms:
```
  Window. getAllWindows().forEach(window => {
      window.hide(); // or window.show();
  });
```

**Comment by sindresorhus** [comment]
@KevinMartin Yes, but I think you just proved why it's worth having an API for this.

**Comment by KevinMartin** [comment]
Yea, I tried implementing for kicks in a gist. Got a bit large. Whats the use for this though? Wouldn't it be bad UX to show the window in a time the user may not want it?

**Comment by sindresorhus** [comment]
> Wouldn't it be bad UX to show the window in a time the user may not want it?

You could say that about most of the APIs though. BrowserWindow has `.hide()` and `.show()` too.

**Comment by zcbenz** [comment]
Closed with #4271.
